src/api-controller.js

```diff
@@ -18,15 +18,35 @@ function _getDefaultHeaders() {
 }
 
 exports.searchIssues = async function (jql) {
-  const res = await fetch(
-    defaultURL + `/rest/api/3/search?maxResults=${process.env.JIRA_MAX_SEARCH_RESULTS}&jql=${jql}`,
-    {
-      method: 'GET',
-      headers: _getDefaultHeaders(),
-      agent: httpsAgent,
+  let formattedRes;
+  let startAt = 0;
+  let pageCount = 1;
+  let issueList = [];
+
+  while (true) {
+    if (pageCount === 0) break;
```

Comment:
rather than waiting for the next iteration, we can break at the end of the loop with this condition

Reply:
```
  const maxResults = process.env.JIRA_MAX_SEARCH_RESULTS || 100;
  let startAt = 0;
  let pageCount = 1;
  let issueList = [];

  while (true) {
    const res = await fetch(defaultURL + `/rest/api/3/search?maxResults=${maxResults}&startAt=${startAt}&jql=${jql}`, {
      method: 'GET',
      headers: _getDefaultHeaders(),
      agent: httpsAgent,
    });

    const data = await res.json();

    if (data.issues) {
      issueList = issueList.concat(data.issues);

      if (startAt === 0) {
        pageCount = Math.ceil(data.total / data.maxResults);
      }

      startAt += data.maxResults;
      pageCount--;

      if (pageCount === 0) break;
    } else {
      throw new Error(data.errorMessages.join('\n'));
    }
  }

  return issueList;
```